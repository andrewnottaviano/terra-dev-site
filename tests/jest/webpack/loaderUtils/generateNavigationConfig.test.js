const path = require('path');
const generatePagesConfig = require('../../../../src/webpack/loaderUtils/generateNavigationConfig');

describe('generateNavigationConfig', () => {
  it('applies all defaults', () => {
    const config = generatePagesConfig({
      siteConfig: {
        additionalSearchDirectories: [],
        primaryNavigationItems: [{
          path: '/test',
          text: 'Test',
          contentExtension: 'ext',
        }],
        sourceFolder: 'tests/jest/webpack/loaderUtils/testContent',
        distributionFolder: 'tests/jest/webpack/loaderUtils/testContent',
        enableDebugLogging: false,
        namespace: 'npm-package',
      },
      resolveExtensions: ['.js'],
      mode: 'development',
      contentDirectory: 'TestA',
      isLernaMonoRepo: false,
      addContextDependency: () => {},
      logger: () => {},
    });
    expect(config.navigationConfig).toMatchSnapshot();
    expect(config.routesMap).toMatchSnapshot();
    expect(config.pageConfig).toMatchSnapshot();
    expect(config.contentImports).toEqual({
      '/test/npm-package/file': path.join(process.cwd(), 'tests/jest/webpack/loaderUtils/testContent/TestA/ext/file.ext.js'),
      '/test/npm-package/file-outside': path.join(process.cwd(), 'tests/jest/webpack/loaderUtils/testContent/TestA/fileOutside.ext.md'),
    });
  });
});
