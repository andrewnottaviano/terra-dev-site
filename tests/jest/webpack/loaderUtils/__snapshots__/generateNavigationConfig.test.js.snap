// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateNavigationConfig test standard config + additional content 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "path": "/test/npm-package/add",
        "text": "Add",
        "type": "md",
      },
      Object {
        "path": "/test/npm-package/file",
        "text": "File",
        "type": "js",
      },
      Object {
        "path": "/test/npm-package/file-outside",
        "text": "File Outside",
        "type": "md",
      },
    ],
    "path": "/test",
    "text": "Test",
  },
]
`;

exports[`generateNavigationConfig test standard config + additional content 2`] = `
Object {
  "/": "/test/npm-package/add",
  "/test": "/test/npm-package/add",
}
`;

exports[`generateNavigationConfig test standard config + additional content 3`] = `
Object {
  "/test/npm-package/add": Object {
    "text": "Add",
    "type": "md",
  },
  "/test/npm-package/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/file-outside": Object {
    "text": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "path": "/test/npm-package/file",
        "text": "File",
        "type": "js",
      },
      Object {
        "path": "/test/npm-package/file-outside",
        "text": "File Outside",
        "type": "md",
      },
    ],
    "path": "/test",
    "text": "Test",
  },
]
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 2`] = `
Object {
  "/": "/test/npm-package/file",
  "/test": "/test/npm-package/file",
}
`;

exports[`generateNavigationConfig tests additionalSearchDirectories 3`] = `
Object {
  "/test/npm-package/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/file-outside": Object {
    "text": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests multiple primary nav items 1`] = `
Array [
  Object {
    "pageConfig": Object {
      "path": "/test/npm-package/file",
      "text": "File",
      "type": "js",
    },
    "path": "/test",
    "text": "Test",
  },
  Object {
    "pageConfig": Object {
      "path": "/doc/npm-package/file",
      "text": "File",
      "type": "js",
    },
    "path": "/doc",
    "text": "Doc",
  },
]
`;

exports[`generateNavigationConfig tests multiple primary nav items 2`] = `
Object {
  "/": "/test/npm-package/file",
  "/doc": "/doc/npm-package/file",
  "/test": "/test/npm-package/file",
}
`;

exports[`generateNavigationConfig tests multiple primary nav items 3`] = `
Object {
  "/doc/npm-package/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/file": Object {
    "text": "File",
    "type": "js",
  },
}
`;

exports[`generateNavigationConfig tests prod mode 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "path": "/test/npm-package/file",
        "text": "File",
        "type": "js",
      },
      Object {
        "path": "/test/npm-package/file-outside",
        "text": "File Outside",
        "type": "md",
      },
    ],
    "path": "/test",
    "text": "Test",
  },
]
`;

exports[`generateNavigationConfig tests prod mode 2`] = `
Object {
  "/": "/test/npm-package/file",
  "/test": "/test/npm-package/file",
}
`;

exports[`generateNavigationConfig tests prod mode 3`] = `
Object {
  "/test/npm-package/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/file-outside": Object {
    "text": "File Outside",
    "type": "md",
  },
}
`;

exports[`generateNavigationConfig tests sorting 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "path": "/test/npm-package/first-file",
        "text": "First File",
        "type": "js",
      },
      Object {
        "path": "/test/npm-package/second-file",
        "text": "Second File",
        "type": "js",
      },
      Object {
        "children": Array [
          Object {
            "path": "/test/npm-package/dir/file",
            "text": "File",
            "type": "js",
          },
        ],
        "text": "Dir",
      },
      Object {
        "path": "/test/npm-package/a",
        "text": "A",
        "type": "js",
      },
      Object {
        "children": Array [
          Object {
            "path": "/test/npm-package/b/file",
            "text": "File",
            "type": "js",
          },
        ],
        "text": "B",
      },
    ],
    "path": "/test",
    "text": "Test",
  },
]
`;

exports[`generateNavigationConfig tests sorting 2`] = `
Object {
  "/": "/test/npm-package/first-file",
  "/test": "/test/npm-package/first-file",
}
`;

exports[`generateNavigationConfig tests sorting 3`] = `
Object {
  "/test/npm-package/a": Object {
    "text": "A",
    "type": "js",
  },
  "/test/npm-package/b/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/dir/file": Object {
    "text": "File",
    "type": "js",
  },
  "/test/npm-package/first-file": Object {
    "text": "First File",
    "type": "js",
  },
  "/test/npm-package/second-file": Object {
    "text": "Second File",
    "type": "js",
  },
}
`;
